samtranslator/utils/utils.py

```diff
@@ -21,3 +21,27 @@ def insert_unique(xs: Any, vs: Any) -> List[Any]:
             xs.append(v)
 
     return cast(List[Any], xs)  # mypy doesn't recognize it
+
+
+class InvalidValueType(Exception):
+    def __init__(self, relative_path: str) -> None:
+        self.relative_path = relative_path
+
+    def full_path(self, path_prefix: str) -> str:
+        if not self.relative_path:
+            return path_prefix
+        return path_prefix + "." + self.relative_path
+
+
+def dict_deep_get(d: Any, path: str) -> Optional[Any]:
+    relative_path = ""
+    _path_nodes = path.split(".")
+    while _path_nodes:
+        if d is None:
+            return None
+        if not isinstance(d, dict):
```

Comment:
Does this enforce the return type? If yes it would be clearer explicitly specified by caller.

Reply:
Also unit tests for this would be great if it's to become the generic way to get stuff out of dicts. It's not clear reading the code how it fails.

Reply:
> Does this enforce the return type? If yes it would be clearer explicitly specified by caller.
similar to `dict.get()` it doesn't enforce the return value. 

> Also unit tests for this would be great if it's to become the generic way to get stuff out of dicts. It's not clear reading the code how it fails.

yeah that's good idea, will add soon

Reply:
it's validating the parents